--- glibc_base/include/libc-symbols.h	2025-03-21 11:51:49.470014870 +0100
+++ glibc_patch/include/libc-symbols.h	2025-03-21 12:55:14.134898488 +0100
@@ -72,10 +72,10 @@
    itself is being compiled, or just some generator program.  */
 #define _LIBC	1
 
-/* Some files must be compiled with optimization on.  */
-#if !defined __ASSEMBLER__ && !defined __OPTIMIZE__
-# error "glibc cannot be compiled without optimization"
-#endif
+///* Some files must be compiled with optimization on.  */
+//#if !defined __ASSEMBLER__ && !defined __OPTIMIZE__
+//# error "glibc cannot be compiled without optimization"
+//#endif
 
 /* -ffast-math cannot be applied to the C library, as it alters the ABI.
    Some test components that use -ffast-math are currently not part of
--- glibc_base/include/alloc_buffer.h	2025-03-21 11:51:48.650029553 +0100
+++ glibc_patch/include/alloc_buffer.h	2025-03-21 20:53:02.564966351 +0100
@@ -83,6 +83,10 @@
 #include <stdlib.h>
 #include <sys/param.h>
 
+#undef _FORTIFY_SOURCE
+
+#define __builtin_constant_p_bypass() (true)
+
 /* struct alloc_buffer objects refer to a region of bytes in memory of a
    fixed size.  The functions below can be used to allocate single
    objects and arrays from this memory region, or write to its end.
@@ -211,7 +215,7 @@
 static __always_inline size_t
 __alloc_buffer_assert_size (size_t size)
 {
-  if (!__builtin_constant_p (size))
+  if (!__builtin_constant_p_bypass (size))
     {
       __errordecl (error, "type size is not constant");
       error ();
@@ -229,7 +233,7 @@
 static __always_inline size_t
 __alloc_buffer_assert_align (size_t align)
 {
-  if (!__builtin_constant_p (align))
+  if (!__builtin_constant_p_bypass (align))
     {
       __errordecl (error, "type alignment is not constant");
       error ();
--- glibc_base/include/features.h	2025-03-21 11:51:49.314017664 +0100
+++ glibc_patch/include/features.h	2025-03-21 20:14:02.778773317 +0100
@@ -18,6 +18,8 @@
 #ifndef	_FEATURES_H
 #define	_FEATURES_H	1
 
+#undef _FORTIFY_SOURCE
+
 /* These are defined by the user (or the compiler)
    to specify the desired environment:
 
